### PR TITLE
feat(migrate): add back `getDatabaseVersion`

### PR DESCRIPTION
**Pull request by jkomyno**
This PR closes https://github.com/prisma/prisma-private/issues/203.
Related Rust PR: https://github.com/prisma/prisma-engines/pull/4071.
Example fake error report with `getDatabaseVersion`: https://prisma-errors.netlify.app/report/14837.

Screenshot 2023-07-14 at 22 39 10

## Notes for the reviewers

The tests in this PR have been updated by locally tampering with the `schema-engine` binary, so that the release version of the binary built from https://github.com/prisma/prisma-engines/pull/4071 was used.

Hence, please:
- first, merge https://github.com/prisma/prisma-engines/pull/4071
- then, rebase this PR with `main` after the pipeline pulls in the Rust changes above
- then, wait for the TS CI to kick in and proceed with the review

This should be easier / cheaper than waiting for integration versions.

### CODE REVIEW

File: packages/internals/src/migrateTypes.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace MigrateTypes {
```
Comment (Jolg42) [inline packages/internals/src/migrateTypes.ts:2]: Nit: technically, I think it's better to name like "SchemaEngineTypes", but not blocking this PR for sure

**Review by Jolg42 — APPROVED** [review]
Nice!

~~Maybe quickly check that the error reporting backend gets the database version like expected~~

File: packages/internals/src/migrateTypes.ts (hunk 1 above)
Comment (jkomyno) [reply packages/internals/src/migrateTypes.ts:2]: I see where you're coming from, and in principle, I'd agree with you. However, considering we still have the `@prisma/migrate` package, wouldn't that be confusing to have conflicting naming conventions within the same codebase?

**Comment by jkomyno** [comment]
> Nice!
> 
> Maybe quickly check that the error reporting backend gets the database version like expected

Tested already! That's why I've mentioned the example fake report https://prisma-errors.netlify.app/report/14837 in this PR's description :)

**Comment by jkomyno** [comment]
> Nice!
> 
> Maybe quickly check that the error reporting backend gets the database version like expected

Tested already! That's why I've mentioned the example fake report https://prisma-errors.netlify.app/report/14837 in this PR's description :)

**Pull request merged**